org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/GradleBuildSupport.java

```diff
@@ -79,7 +79,11 @@ public void update(IProject project, boolean force, IProgressMonitor monitor) th
 				BuildConfiguration buildConfiguration = GradleProjectImporter.getBuildConfiguration(Paths.get(projectPath));
 				gradleBuild = GradleCore.getWorkspace().createBuild(buildConfiguration);
 			}
-			if (isRoot) {
+			File buildFile = project.getFile(GradleProjectImporter.BUILD_GRADLE_DESCRIPTOR).getLocation().toFile();
+			File settingsFile = project.getFile(GradleProjectImporter.SETTINGS_GRADLE_DESCRIPTOR).getLocation().toFile();
+			boolean shouldUpdate = (buildFile.exists() && JavaLanguageServerPlugin.getDigestStore().updateDigest(buildFile.toPath()))
+					|| (settingsFile.exists() && JavaLanguageServerPlugin.getDigestStore().updateDigest(settingsFile.toPath()));
+			if (force || isRoot || shouldUpdate) {
```

Comment:
When vmInstall or gradle home changed, the `force` flag will be true. Is it necessary to update all projects?

Reply:
Yes, it is. See https://github.com/eclipse/eclipse.jdt.ls/blob/master/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/MavenBuildSupport.java#L89

Reply:
I'm not worried about this commit can fix the outdated dependencies issue. But this fix looks like in conflict with your previous PR https://github.com/eclipse/eclipse.jdt.ls/pull/1467.

When i tried this commit in springboot, and changed the user setting "java.import.gradle.java.home", i see lots of synchronization jobs are triggered.
![image](https://user-images.githubusercontent.com/14052197/100699231-59a57b80-33d5-11eb-9775-81242f55f9de.png)

Reply:
> When i tried this commit in springboot, and changed the user setting "java.import.gradle.java.home", i see lots of synchronization jobs are triggered.

They have to be synchronized.

Reply:
Could you please explain the reason? I assume when you're synchronizing the gradle root project, all projects under it will be updated and no need to synchronize each subproject again. Please correct me if it's not right.

Reply:
You are right. I have updated the PR.